### PR TITLE
[AAP-8171] Add related groups list to group details

### PR DESCRIPTION
**Pull request by matoval**
Jira ticket: https://issues.redhat.com/browse/AAP-8171 & https://issues.redhat.com/browse/AAP-8314

Add related groups list to group details

Before:
image

After:
image

### CODE REVIEW

File: frontend/awx/main/routes/useAwxInventoryRoutes.tsx [hunk 1]
```diff
@@ -119,6 +121,11 @@ export function useAwxInventoryRoutes() {
           path: ':inventory_type/:id/host/:host_id/edit',
           element: <EditHost />,
         },
+        {
+          id: AwxRoute.InventoryGroupRelatedGroupsCreate,
+          path: ':inventory_type/:id/groups/:group_id/related_groups/add',
```
Comment (prat98) [inline frontend/awx/main/routes/useAwxInventoryRoutes.tsx:126]: Small change to match the route specified in [AAP-8314](https://issues.redhat.com/browse/AAP-8314) and the old UI
```suggestion
          path: ':inventory_type/:id/groups/:group_id/nested_groups/add',
```

**Review by prat98 — COMMENTED** [review]
These changes look great! 

Able to validate the subset of acceptance criteria laid out in [AAP-19321](https://issues.redhat.com/browse/AAP-19321) were met. 

Ran component tests locally successfully. 

Just a small change suggested around path to add related groups 

Also having my local cypress crash when attempting to run inventories E2E tests locally. Not sure if that's related to your changes though

**Comment by matoval** [comment]
> @matoval is there a reason not to use the existing Groups list here?

There's no linked line of code but I'm assuming you are talking about the related groups list? If so I didn't use the exist groups list there because the actions in the toolbar and row are different.

File: frontend/awx/resources/groups/hooks/useRelatedGroupsToolbarActions.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                <GroupSelectDialog
+                  groupId={params.group_id as string}
+                  onSelectedGroups={onSelectedGroups}
+                />
+              ),
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('New group'),
```
Comment (AlexSCorey) [inline frontend/awx/resources/groups/hooks/useRelatedGroupsToolbarActions.tsx:92]:
```suggestion
            label: t('Create new group'),
```

File: frontend/awx/resources/groups/hooks/useRelatedGroupsToolbarActions.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        selection: PageActionSelection.None,
+        variant: ButtonVariant.primary,
+        isPinned: true,
+        icon: PlusIcon,
+        label: t('Add group'),
+        actions: [
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('Existing group'),
```
Comment (AlexSCorey) [inline frontend/awx/resources/groups/hooks/useRelatedGroupsToolbarActions.tsx:80]:
```suggestion
            label: t('Add existing group'),
```

**Review by AlexSCorey — APPROVED** [review]
This looks good to me.  Well done!

**Pull request merged**